### ISSUE TITLE
Extraction generates bad Ocaml

### ISSUE THREAD

**Issue by Zdancewic** [DeepSpec/InteractionTrees#19]
I think that it's probably a bug in Coq's extraction mechanism, but when I try to extract the ITree.ml file, I get a type-mismatch in the signature generated in ITree.mli.    The problem seems to be something to do with Lazy.t, or perhaps the universe polymorphism in ext-lib, and appears in all three of the typeclasses (Applicative, Functor, and Monad) defined in ITree.v.

(For now, I've just commented out those declarations, but we should find a better fix.)

**Comment by Zdancewic** [comment]
Here's the error:

```
File "ml/extracted/ITree.ml", line 1:
Error: The implementation ml/extracted/ITree.ml
       does not match the interface ml/extracted/iTree.cmi:
       Values do not match:
         val coq_Applicative_itree :
           ('_weak1, '_weak2) __itree Lazy.t Applicative.coq_Applicative
       is not included in
         val coq_Applicative_itree :
           ('a1, __) __itree Lazy.t Applicative.coq_Applicative
       File "ml/extracted/ITree.mli", line 30, characters 0-68:
         Expected declaration
       File "ml/extracted/ITree.ml", line 63, characters 4-25:
         Actual declaration
```

**Comment by Lysxia** [comment]
That's an error caused by OCaml's value restriction. The Applicative and Monad instances try to be polymorphic but they're not syntactically values. We can eta expand to avoid this issue.

**Comment by Lysxia** [comment]
Closed by 5ec74e88bf84bea2157f33358460a003f2499c6a.

**Comment by Zdancewic** [comment]
Thanks!  That fixed it.
